cucumber-expressions/ruby/spec/cucumber/cucumber_expressions/regular_expression_spec.rb

```diff
@@ -46,6 +46,10 @@ module CucumberExpressions
           "I can cancel the 1st slide upload")
         ).to eq(["I", "can", 1, "slide"])
       end
+      
+      it "ignores escaped 2.x parenthesis" do
+        expect( match(/Across the line\(s\)/, 'Across the line\(s\)') ).to be_nil
```

Comment:
This should be:

```ruby
expect( match(/Across the line\(s\)/, 'Across the line(s)') ).to eq([])
```

That will reproduce the bug when the `CAPTURE_GROUP_PATTERN` is as before. Can you push another commit with that fix please?

I'll have a think about what to do with negative lookahead - ideally we'll use the same regexps in all implementations...